### PR TITLE
Bug 1717212 - Compare View query string parameters are not updated

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**
The filter buttons ('Hide uncomparable results', 'Show only important changes', 'Hide uncertain results', 'Show only noise' ) and the filter text from the Compare View Page are not reflected in the URL as they should.

This PR should fix this issue.

### CODE REVIEW

File: tests/ui/perfherder/compare-view/compare_table_test.jsx [hunk 1]
```diff
@@ -204,6 +208,68 @@ test('toggle buttons should filter results by selected filter', async () => {
   expect(result2).toBeInTheDocument();
 });
 
+test('toggle all buttons should update the URL params', async () => {
+  const { getByText } = compareTableControls();
+
+  const showImportant = getByText(filterText.showImportant);
```
Comment (esanuandra) [inline tests/ui/perfherder/compare-view/compare_table_test.jsx:214]: `getByText` is missing `await waitFor()`

File: ui/perfherder/compare/CompareTableControls.jsx [hunk 2]
```diff
@@ -133,6 +142,43 @@ export default class CompareTableControls extends React.Component {
...
+      hideUncertain,
+      showNoise,
+    } = this.state;
+    const {
+      showOnlyComparable,
+      showOnlyImportant,
+      showOnlyConfident,
+      showOnlyNoise,
+      filter,
+    } = parseQueryParams(window.location.search);
```
Comment (esanuandra) [inline ui/perfherder/compare/CompareTableControls.jsx:160]: I think we have all these flags in `this.props.validated`.

Comment (beatrice-acasandrei) [reply ui/perfherder/compare/CompareTableControls.jsx]: I've changed the implementation so that it doesn't need the URL params anymore.

**Review by esanuandra — APPROVED** [review]

**Comment by beatrice-acasandrei** [comment]
Hi @sarah-clements ! Can you please review & merge my PR when you get the chance?

**Review by sarah-clements — APPROVED** [review]

**Pull request merged**